### PR TITLE
Make the UUID mandatory

### PR DESCRIPTION
**Pull request by kleewho**
refactor: Breaking change, uuid parameter is mandatory.

uuid is required parameter in PNConfiguration constructor.

BREAKING CHANGES: SDK will not generate or store uuid automatically. uuid is required parameter in PNConfiguration constructor.

### CODE REVIEW

File: src/main/java/com/pubnub/api/PubNubUtil.java [hunk 1]
```diff
@@ -290,4 +290,10 @@ private static String requestBodyToString(final Request request) {
         return collection == null || collection.isEmpty();
     }
 
+    public static void requirePN(boolean value, PubNubError error) throws PubNubException {
```
Comment (bartk) [inline src/main/java/com/pubnub/api/PubNubUtil.java:293]: maybe just `require()`?

File: src/test/java/com/pubnub/contract/MockPubnubService.kt [hunk 2]
```diff
@@ -15,6 +30,10 @@ interface MockPubnubService {
     fun expect(): Call<ExpectResponse>
 }
 
+class A {
```
Comment (bartk) [inline src/test/java/com/pubnub/contract/MockPubnubService.kt:33]: Do we need that class?

**Review by bartk — CHANGES_REQUESTED** [review]

File: src/main/java/com/pubnub/api/PubNubUtil.java (hunk 1 above)
Comment (kleewho) [reply src/main/java/com/pubnub/api/PubNubUtil.java:293]: done

File: src/test/java/com/pubnub/contract/MockPubnubService.kt (hunk 2 above)
Comment (kleewho) [reply src/test/java/com/pubnub/contract/MockPubnubService.kt]: Nope :)

**Review by bartk — DISMISSED** [review]

**Review by bartk — DISMISSED** [review]

**Comment by kleewho** [comment]
@client-engineering-bot release as v6.0.0

**Review by kazydek — APPROVED** [review]

**Review by bartk — APPROVED** [review]

**Pull request merged**